### ISSUE TITLE
Use `mapCube` to compute YAXArrays

### ISSUE THREAD

**Issue by MartinuzziFrancesco** [awesome-spectral-indices/SpectralIndices.jl#50]
pinging @lazarusA

**Comment by MartinuzziFrancesco** [comment]
julia> using PartialFunctions

julia> function test(::Type{T}, a, b) where {T<:Number}
           return T(a+b)
       end
test (generic function with 1 method)

julia> tt = test $ Float32
test(Float32, ...)

julia> tt(1, 2)
3.0f0

**Comment by MartinuzziFrancesco** [comment]
is `map` multi-threaded already?

**Comment by lazarusA** [comment]
don't enforce type on function args, see error from mapCube:

```
ERROR: TaskFailedException

    nested task error: MethodError: no method matching NDVI_func(::Type{…}, ::SubArray{…}, ::SubArray{…})
```

**Comment by MartinuzziFrancesco** [comment]
addressed in #52
